scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,263 @@
+     */
+    private fun checkProhibitedFileNamePattern(
+      repoPath: String,
+      searchFiles: List<File>,
+      prohibitedFilenameObj: FilenameCheck,
+    ): Boolean {
+      val prohibitedFilenameRegex = prohibitedFilenameObj.getProhibitedFilenameRegex().toRegex()
+      val matchedFiles = searchFiles.filter { file ->
+        return@filter file.name !in prohibitedFilenameObj.getExemptedFileNameList()
+          && prohibitedFilenameRegex.matches(
+          RepoFile.retrieveFilePath(
+            file,
+            repoPath)
+        )
+      }
+
+      logProhibitedFilenameFailure(
+        repoPath = repoPath,
+        matchedFiles = matchedFiles,
+        errorToShow = prohibitedFilenameObj.getFailureMessage()
+      )
+
+      return matchedFiles.toList().size != 0
+    }
+
+    /**
+     * Checks for a prohibited file content.
+     *
+     * @param repoPath the path of the repo
+     * @param searchFiles a list of all the files which needs to be checked
+     * @param prohibitedContentObj proto object of FileContentCheck
+     * @return check failed or passed
+     */
+    private fun checkProhibitedContent(
+      repoPath: String,
+      searchFiles: List<File>,
+      prohibitedContentObj: FileContentCheck,
+    ): Boolean {
+      var contentCheckFailedFlag = false
+      val fileNameRegex = prohibitedContentObj.getFilenameRegex().toRegex()
+      val prohibitedContentRegex = prohibitedContentObj.getProhibitedContentRegex().toRegex()
+      val matchedFiles = searchFiles.filter {
+        it.name !in prohibitedContentObj.getExemptedFileNameList()
+          && fileNameRegex.matches(it.name)
+          && File(it.toString()).bufferedReader().lineSequence()
+          .foldIndexed(initial = false) { lineNumber, acc, lineStr ->
+            val matches = prohibitedContentRegex.matches(lineStr)
+            if (matches) {
+              logProhibitedContentFailure(
+                lineNumber = lineNumber + 1,
+                errorToShow = prohibitedContentObj.getFailureMessage(),
+                filePath = RepoFile.retrieveFilePath(it, repoPath)
+              )
+            }
+            return@foldIndexed acc || matches
+          }
+      }
+      return matchedFiles.size != 0
+    }
+
+
+    /** Logs the failures for filename pattern violation.
+     *
+     * @param repoPath the path of the repo to be analyzed
+     * @param errorToShow the filename error to be logged
+     * @param matchedFiles a list of all the files which had the filenaming violation
+     * @return log the filename failures
+     */
+    private fun logProhibitedFilenameFailure(
+      repoPath: String,
+      errorToShow: String,
+      matchedFiles: List<File>
+    ) {
+      if (matchedFiles.size != 0) {
+        print("Filename pattern violation: $errorToShow\n")
+        matchedFiles.forEach {
+          print("Prohibited file: [ROOT]/${RepoFile.retrieveFilePath(it, repoPath)}\n")
+        }
+        print("\n")
```

Comment:
Does `println()` will work here?
WDYT?

Reply:
Yes, it will work fine. Refactored. PTAL.

Reply:
Also, 

```diff
-  print("Filename pattern violation: $errorToShow\n")
+ println("Filename pattern violation: $errorToShow")

- print("Prohibited file: [ROOT]/${RepoFile.retrieveFilePath(it, repoPath)}\n")
+ println("Prohibited file: [ROOT]/${RepoFile.retrieveFilePath(it, repoPath)}")
```

Reply:
Done.